contract/views/account_analytic_account_view.xml

```diff
@@ -95,6 +98,51 @@
         </field>
     </record>
 
+    <record id="account_analytic_account_sale_form" model="ir.ui.view">
+        <field name="name">account.analytic.account.sale.form</field>
+        <field name="model">account.analytic.account</field>
+        <field name="inherit_id" ref="account_analytic_account_recurring_form_form"/>
+        <field name="mode">primary</field>
+        <field name="priority" eval="20"/>
+        <field name="arch" type="xml">
+            <field name="partner_id" position="attributes">
+                <attribute name="string">Customer</attribute>
+                <attribute name="domain">[('customer', '=', True)]</attribute>
+                <attribute name="context">{'default_customer': True, 'default_supplier': False}</attribute>
+            </field>
+            <field name="journal_id" position="attributes">
+                <attribute name="domain">[('type', '=', 'sale')]</attribute>
+            </field>
+            <field name="product_id" position="attributes">
+                <attribute name="domain">[('sale_ok', '=', True)]</attribute>
+            </field>
+        </field>
+    </record>
+
+    <record id="account_analytic_account_purchase_form" model="ir.ui.view">
+        <field name="name">account.analytic.account.purchase.form</field>
+        <field name="model">account.analytic.account</field>
+        <field name="inherit_id" ref="account_analytic_account_recurring_form_form"/>
+        <field name="mode">primary</field>
+        <field name="priority" eval="20"/>
+        <field name="arch" type="xml">
+            <field name="partner_id" position="attributes">
+                <attribute name="string">Supplier</attribute>
```

Comment:
Vendor for being aligned with new Accounting nomenclature?

Reply:
But in all the module I used the Supplier nomenclature...